tensorboard/plugins/base_plugin.py

```diff
@@ -69,14 +74,32 @@ def is_active(self):
     raise NotImplementedError()
 
 
-class TBContext(collections.namedtuple('TBContext', ('logdir', 'multiplexer'))):
+class TBContext(object):
   """Magic container of information passed from TensorBoard core to plugins.
 
-  A TBContext instance is passed to the constructor of a TBPlugin class.
-
-  Fields:
-    logdir: The string logging directory TensorBoard was started with.
-    multiplexer: An EventMultiplexer with underlying TB data, or None if SQL
-        mode is being used.
+  A TBContext instance is passed to the constructor of a TBPlugin class. Plugins
+  are strongly encouraged to assume that any of these fields can be None.
```

Comment:
If one of these fields is `None` but the plugin needs it, what should it do? Perhaps it should do something like raise a `ValueError` at instantiation time—in which case the TB framework should be handling these and dropping the plugin instead of crashing the app.